controllers/controllers/workloads/shared.go

```diff
@@ -38,29 +44,173 @@ func setStatusConditionOnLocalCopy(conditions *[]metav1.Condition, conditionType
 	})
 }
 
-func createSubnamespaceAnchor(ctx context.Context, client client.Client, req ctrl.Request, object client.Object, labels map[string]string) (v1alpha2.SubnamespaceAnchor, error) {
-	anchor := v1alpha2.SubnamespaceAnchor{
+func createOrPatchNamespace(ctx context.Context, client client.Client, log logr.Logger, object client.Object, labels map[string]string) error {
+	namespace := &corev1.Namespace{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      req.Name,
-			Namespace: req.Namespace,
-			Labels:    labels,
-			OwnerReferences: []metav1.OwnerReference{
-				{
-					APIVersion: APIVersion,
-					Kind:       object.GetObjectKind().GroupVersionKind().Kind,
-					Name:       object.GetName(),
-					UID:        object.GetUID(),
-				},
-			},
+			Name: object.GetName(),
+		},
+	}
+
+	result, err := controllerutil.CreateOrPatch(ctx, client, namespace, func() error {
+		if namespace.ObjectMeta.Labels == nil {
```

Comment:
nitpick: no need to use `ObjectMeta` in `namespace.ObjectMeta.Labels`, `namespace.Labels` is just fine. Same goes withe the rest of the meta fields: name, annotations, etc.

Reply:
Good point. We took these out of this commit, but we use `ObjectMeta.*` in many places throughout the codebase and we decided not to touch existing code.